modules/redhen_connection/src/ConnectionHtmlRouteProvider.php

```diff
@@ -100,31 +105,92 @@ protected function getAddFormRoute(EntityTypeInterface $entity_type) {
     }
   }
 
+
   /**
-   * Gets the add page route.
+   * Gets all entity type add connection routes.
    *
    * @param \Drupal\Core\Entity\EntityTypeInterface $entity_type
    *   The entity type.
    *
-   * @return \Symfony\Component\Routing\Route|null
-   *   The generated route, if available.
+   * @return array
+   *   An array of routes, if available.
    */
-  protected function getAddPageRoute(EntityTypeInterface $entity_type) {
-    $route = new Route("/redhen/{redhen_type}/{entity}/connection/add");
-    $parameters = [
-      'redhen_type' => ['type' => 'redhen_type'],
-      'entity' => ['type' => 'entity:{redhen_type}'],
-    ];
-    $route
-      ->setOption('parameters', $parameters);
-    $route
-      ->setDefaults([
-        '_controller' => 'Drupal\redhen_connection\Controller\ConnectionAddController::add',
-        '_title' => "Add {$entity_type->getLabel()}",
-      ])
-      ->setRequirement('_entity_create_access', $entity_type->id());
-
-    return $route;
+  protected function getAddEntityPageRoutes($entity_type) {
+    $routes = [];
+
+    $entity_type_id = $entity_type->id();
+
+    $entity_types = \Drupal::entityTypeManager()->getDefinitions();
+
+    foreach ($entity_types as $type) {
+      if ($canonical = $type->getLinkTemplate('canonical')) {
+        $type_id = $type->id();
+
+        if ($type_id == 'redhen_connection_type' || $type_id == 'redhen_connection_role') {
+          continue;
+        }
+
+        $parameters = [
+          $type_id => ['type' => 'entity:'. $type_id],
+        ];
+
+        $route = new Route($canonical . "/connection/add");
+        $route->setOption('parameters', $parameters);
+        $route
+          ->setDefaults([
+            '_controller' => 'Drupal\redhen_connection\Controller\ConnectionAddController::add',
+            '_title' => 'Add ' . $entity_type->getLabel(),
+          ])
+          ->setOption('_admin_route', TRUE)
+          ->setRequirement('_entity_create_access', $entity_type_id);
```

Comment:
The foreach here is iterating over each entity type and looking for anything with a canonical link template, then it adds a route for each entity with the url pattern `$canonical_path/connection/add` mapped to the ConnectionAddController::add method. 

Yes that`$entity_type` will always just be `redhen_connection`.  I think I might have just passed that variable along because it was used in the prior logic. but I realize that the title I am setting isn't really useful. Seems like it could change to `Add $type->getLabel() Connection`. And I think I can remove that $entity_type variable altogether, since we know what it should be, and it will hopefully make this a bit more readable. 

Combined both functions into one to make it DRY-er.